### PR TITLE
Infrastructure

### PR DESCRIPTION
**Pull request by ajax-malov-n**
Added : 
 + DockerFile
 + Docker Compose
 + K8s
Changed:
 + Github Actions

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -26,8 +26,10 @@ repositories {
 }
 
 dependencies {
+    implementation("org.springframework.boot:spring-boot-starter-actuator")
```
Comment (ajax-gnatiuk-s) [inline build.gradle.kts]: move closer to another spring deps

File: k8s-configuration/stockanalyzer-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: stockanalyzer-configmap
+data:
+  symbols: "SPUP,BNRG,HIG.PRG,RILYT,GBARF,WINN,PWWBF,RHBAF,CLVR,UPLD,AJGFF,HLLY,MSIU,ALNKF,UCRD,MGNT,IMAQ,SWRBF,BAYRY,RVHCS,APTPF,LALFS,HIVE,FNLIF,TLIS,CRTO,ORGO,OZBKF,UNIKF,DSPHF,FRCLL,CLMB,NCXS,GXRFF,JL,AMSRF,BITC,TLSNY,TGMR,TZPC,ADDHY,CBVTF,GIGM,CIBR,AFSIM,GULDF,CSBF,LFMDP,ABMC,LVWR.WS,PDEX,BFS.PRE"
```
Comment (ajax-gnatiuk-s) [inline k8s-configuration/stockanalyzer-configmap.yaml]: why it's the part of k8s configuration. It looks like we should move it to `application.yaml`

File: k8s-configuration/stockanalyzer.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: stockanalyzer
+  labels:
+    app: stockanalyzer
+spec:
+  replicas: 1
```
Comment (ajax-gnatiuk-s) [inline k8s-configuration/stockanalyzer.yaml:8]: let's have at least `2`?

Comment (ajax-malov-n) [reply k8s-configuration/stockanalyzer.yaml:8]: I cannot just do it. My service uses third-party API and I can only fetch 50 records/min

Comment (ajax-shulepa-a) [reply k8s-configuration/stockanalyzer.yaml:8]: it looks like you need a distributed lock for your task. I expect that after introducing mongo you'll fix this
https://github.com/lukas-krecan/ShedLock

Comment (ajax-malov-n) [reply k8s-configuration/stockanalyzer.yaml:8]: I will check 'ShedLock' and add it with mongo migration. Okay?

File: detekt.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+style:
+  VarCouldBeVal:
+    active: false
```
Comment (ajax-shulepa-a) [inline detekt.yml]: disable it only for test files

File: docker-compose.yml [hunk 5]
```diff
@@ -2,29 +2,40 @@ version: "3.8"
...
+      - MONGO_PASSWORD=${MONGO_ROOT_PASSWORD}
+      - MONGO_USERNAME=${MONGO_ROOT_USERNAME}
       - FINNHUB_API_TOKEN=${FINNHUB_API_TOKEN}
       - SYMBOLS=${SYMBOLS}
+    env_file:
+      - .env
 
   mongo:
-    image: mongo
+    image: mongo:8.0.0-rc19-noble
```
Comment (ajax-shulepa-a) [inline docker-compose.yml]: use stable version, please

File: k8s-configuration/mongo.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      app: mongodb
+  template:
+    metadata:
+      labels:
+        app: mongodb
+    spec:
+      containers:
+        - name: mongodb
+          image:
+            mongo:8.0.0-rc19-noble
```
Comment (ajax-shulepa-a) [inline k8s-configuration/mongo.yaml]: stable version

File: detekt.yml (hunk 4 above)
Comment (ajax-malov-n) [reply detekt.yml]: This warning occurs not only in test files

**Review by ajax-shulepa-a — APPROVED** [review]

File: src/main/resources/application-local.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+spring:
+  application:
+    name: stockanalyzer
+  data:
+    mongodb:
+      uri: mongodb://root:secret@localhost:27017/stocks?authSource=admin
+api:
+  finnhub:
+    token: ${FINNHUB_API_TOKEN}
+    symbols: SPUP,BNRG,HIG.PRG,RILYT,GBARF,WINN,PWWBF,RHBAF,CLVR,UPLD,AJGFF,HLLY,MSIU,ALNKF,UCRD,MGNT,IMAQ,SWRBF,BAYRY,RVHCS,APTPF,LALFS,HIVE,FNLIF,TLIS,CRTO,ORGO,OZBKF,UNIKF,DSPHF,FRCLL,CLMB,NCXS,GXRFF,JL,AMSRF,BITC,TLSNY,TGMR,TZPC,ADDHY,CBVTF,GIGM,CIBR,AFSIM,GULDF,CSBF,LFMDP,ABMC,LVWR.WS,PDEX,BFS.PRE
```
Comment (ajax-gnatiuk-s) [inline src/main/resources/application-local.yaml]: it duplicates property from `application.yaml`. 
Maybe, we should delete `symbols` here?

**Review by ajax-gnatiuk-s — APPROVED** [review]

File: detekt.yml (hunk 4 above)
Comment (ajax-gnatiuk-s) [reply detekt.yml]: example for non-tests files?

Comment (ajax-malov-n) [reply detekt.yml]: In Config file lateinit:
```kotlin
  @Value("\${api.finnhub.token}")
    private lateinit var apiToken: String
```

Comment (ajax-gnatiuk-s) [reply detekt.yml]: so, u use field injection. 
Let's refactor?

**Review by ajax-samondros-b — APPROVED** [review]

**Pull request merged**